### PR TITLE
infrastructure for running completion tests for old and new completions

### PR DESCRIPTION
**Pull request by ivanbasov**

### CODE REVIEW

File: src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb [hunk 1]
```diff
@@ -37,48 +43,52 @@ End Class]]></document>
             End Using
         End Function
 
-        <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
-        Public Async Function SnippetExpansionNoteNotAddedToDescription_ShortcutIsProperSubstringOfInsertedText() As Task
-            Using state = CreateVisualBasicSnippetExpansionNoteTestState(_markup, "Interfac")
+        <WpfTheory, Trait(Traits.Feature, Traits.Features.Completion)>
```
Comment (sharwell) [inline src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb]: ❕ Missing member data

File: src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb [hunk 2]
```diff
@@ -18,17 +18,23 @@ Class Goo
     $$
 End Class]]></document>
 
-        <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
-        Public Async Function ColonDoesntTriggerSnippetInTupleLiteral() As Task
-            Using state = CreateVisualBasicSnippetExpansionNoteTestState(_markup, "Interface")
+        Private Shared Function GetAllCompletions() As IEnumerable(Of Object())
```
Comment (sharwell) [inline src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb]: :bulb: Make these read-only properties named `AllCompletions`

File: src/EditorFeatures/TestUtilities2/Intellisense/Completions.vb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
+    Public Enum Completions
```
Comment (sharwell) [inline src/EditorFeatures/TestUtilities2/Intellisense/Completions.vb]: :bulb: `CompletionImplementation` (rename and non-plural)

**Comment by sharwell** [comment]
@ivanbasov If/when you merge **master** to **asynccompletion** and need to disable one of the tests provided in the master branch, simply replace this:

```vb
<MemberData(NameOf(GetAllCompletions))>
```

With this:

```vb
<InlineData(CompletionImplementation.Legacy)>
<InlineData(CompletionImplementation.Modern, Skip:="https://github.com/dotnet/roslyn/issues/#####")>
```

File: src/EditorFeatures/TestUtilities2/Intellisense/Completions.vb [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+        OldCompletion = 1
```
Comment (sharwell) [inline src/EditorFeatures/TestUtilities2/Intellisense/Completions.vb]: :bulb: In this branch, suggest the name `Legacy`, and in the **asynccompletion** branch add the value `Modern`

File: src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb (hunk 1 above)
Comment (ivanbasov) [reply src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb]: Thank you!

---
In reply to: [208790050](https://github.com/dotnet/roslyn/pull/29176#discussion_r208790050) [](ancestors = 208790050)

File: src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb (hunk 2 above)
Comment (ivanbasov) [reply src/VisualStudio/Core/Test/Completion/VisualBasicCompletionSnippetNoteTests.vb]: AllCompletionImplementations

---
In reply to: [208790208](https://github.com/dotnet/roslyn/pull/29176#discussion_r208790208) [](ancestors = 208790208)

**Comment by ivanbasov** [comment]
Sure, Thank you, @sharwell!

---
In reply to: [411617447](https://github.com/dotnet/roslyn/pull/29176#issuecomment-411617447) [](ancestors = 411617447)

**Pull request merged**